ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```diff
@@ -0,0 +1,885 @@
+        },
+        "timestampColumn" : {
+          "$ref" : "#/definitions/TimestampColumn"
+        },
+        "kafkaTopicName" : {
+          "type" : "string"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "windowInfo" : {
+          "$ref" : "#/definitions/WindowInfo"
+        }
+      },
+      "title" : "createTableV1",
+      "required" : [ "@type", "sourceName", "schema", "kafkaTopicName", "formats" ]
+    },
+    "RegisterTypeCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "registerTypeV1" ],
+          "default" : "registerTypeV1"
+        },
+        "type" : {
+          "type" : "string"
+        },
+        "name" : {
+          "type" : "string"
+        }
+      },
+      "title" : "registerTypeV1",
+      "required" : [ "@type", "type", "name" ]
+    },
+    "DropSourceCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "dropSourceV1" ],
+          "default" : "dropSourceV1"
+        },
+        "sourceName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "dropSourceV1",
+      "required" : [ "@type", "sourceName" ]
+    },
+    "DropTypeCommand" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "dropTypeV1" ],
+          "default" : "dropTypeV1"
+        },
+        "typeName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "dropTypeV1",
+      "required" : [ "@type", "typeName" ]
+    },
+    "QueryPlan" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "sources" : {
+          "type" : "array",
+          "items" : {
+            "type" : "string"
+          }
+        },
+        "sink" : {
+          "type" : "string"
+        },
+        "queryId" : {
+          "type" : "string"
+        },
+        "physicalPlan" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        }
+      },
+      "required" : [ "sources", "sink", "queryId", "physicalPlan" ]
+    },
+    "StreamAggregate" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamAggregateV1" ],
+          "default" : "streamAggregateV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "nonFuncColumnCount" : {
+          "type" : "integer"
+        },
```

Comment:
Feels wrong to be serializing this.

I'd of thought the parent step defines the set of non-aggregate columns...

Reply:
I'm not sure here. It seems reasonable to specify which columns are result columns and which columns are aggregation parameters. This isn't necessarily determined by the grouping columns.

For example, we could have something like `SELECT X, LEN(X), COUNT(*) GROUP BY X;`. In this case, there are columns in the projection that aren't being explicitly grouped on.

That said, maybe a column count isn't the best way...

Reply:
This seems like a clear example of 'implementation bleeding into the schema'.

I still have a feeling this isn't needed and be be inferred from other info..

Reply:
@big-andy-coates it makes sense to me to have the aggregation step specify which columns should be retained in the output. I agree that the first n columns in the input isn't a great way of specifying this as it assumes the input is structured a specific way. What makes more sense to me is to have the step specify a list of column references. So if the input schema is:
`X INT, Y INT, Z INT`, and we want the result to have `X`, `Z`, and the sum of `Y`, then the aggregate step would specify `[X, Z]` as the list of column references to retain, and `[SUM(Y)]` for the aggregate function(s). wdyt?